clicommand/agent_start.go

```diff
@@ -214,13 +214,13 @@ var AgentStartCommand = cli.Command{
 		},
 		cli.BoolFlag{
 			Name:   "disconnect-after-job",
-			Usage:  "Disconnect the agent after running a job",
+			Usage:  "Disconnect the agent as soon as the job they're running completes",
 			EnvVar: "BUILDKITE_AGENT_DISCONNECT_AFTER_JOB",
 		},
 		cli.IntFlag{
 			Name:   "disconnect-after-idle-timeout",
 			Value:  0,
-			Usage:  "If no jobs have come in for the specified number of seconds, disconnect the agent",
+			Usage:  "If no jobs have come in for the specified number of seconds, disconnect the agent. Note that the default for this flag is 0, and means \"don't disconnect idle agents\". If you want to disconnect agents as soon as they become idle, use the --disconnect-after-job flag instead",
```

Comment:
I personally would lean towards brevity:

"The maximum idle time in seconds to wait for a job before disconnecting. The default of 0 means no timeout."

Reply:
This one carries the same extra questions about how it behaves in `--spawn` mode.

Reply:
this one looks to be designed with `--spawn` in mind, where `--disconnect-after-job` isn't. when `--disconnect-after-idle-timeout` is specified, the agent will only disconnect if _all_ of the workers are idle - see [agent/agent_worker.go:234](https://github.com/buildkite/agent/blob/main/agent/agent_worker.go#L234=)